dolosse/generic_json_consumer.py

```diff
@@ -0,0 +1,153 @@
+    arguments = parser.parse_args()
+
+    # To consume latest messages and auto-commit offsets
+    return arguments.topic, arguments.group, arguments.server
+
+def redraw(figures):
+    """Redraw all figures from a dict of figures"""
+    for key in figures:
+        figures[key].canvas.draw()
+        figures[key].canvas.flush_events()
+
+
+def update_data(data, value):
+    """Update the list of data, given some json input"""
+    try:
+        update = loads(value)
+    except decoder.JSONDecodeError as problem:
+        print("Error: Flawed JSON in value: ",
+              value)
+        print("Error message: ", problem)
+        update = ""
+    data.update(update)
+
+
+def graph(topic, group, server):
+    """Main function - reads json data from a kafka topic and graphs it"""
+
+    screen = initscr()
+    cbreak()
+    screen.nodelay(True)
+    screen.getch()
+
+    print("Generic Graphing Consumer")
+    print("Press any key to exit...")
+
+    data = {}
+    line = {}
+    plots = {}
+    figures = {}
+    axes = {}
+    count = 0
+    pyplot.ion() #Interactive mode on
+    configuration = {"bootstrap.servers": server,
+                     "group.id": group,
+                     "default.topic.config" : {
+                         "auto.offset.reset": "earliest"},
+                     "enable.auto.commit": False}
+    consumer = Consumer(configuration)
+    consumer.subscribe([topic])
+    keyboard_input = ERR
+
+    while True:
+        if keyboard_input != ERR:
+            break
+        keyboard_input = screen.getch()
+        message = consumer.consume(num_messages=1, timeout=0.2)
+        if (message is None) or (message == []):
+            redraw(figures)
+        elif message[0].error():
+            print("Something went wrong with the read!")
+            print(message.error())
+        else:
+            update_data(data, message[0].value())
+            count += 1
+
+            populate(line, data, count)
+
+            for key in line:
+                while len(line[key]) < count:
+                    line[key].append(0) #Variable went missing from data.
+                    #Buffer with zeroes for now.
+                if key in plots:
+                    axes[key][0].set_ydata(line[key])
+                    axes[key][0].set_xdata(list(range(count)))
+                    plots[key].relim()
+                    plots[key].autoscale_view()
+                else:
+                    figures[key], plots[key] = pyplot.subplots()
+                    axes[key] = plots[key].plot(line[key])
+                    plots[key].set_autoscaley_on(True)
+                    plots[key].set_autoscalex_on(True)
+                    plots[key].set_title(key)
+
+            redraw(figures)
+
+    nocbreak()
+    endwin()
+
+
+def main_args():
+    """Grabs parameters from the arguments"""
+    topic_arg, group_arg, server_arg = get_args()
+    graph(topic_arg, group_arg, server_arg)
+
+
+if __name__ == '__main__':
+    main_args()
```

Comment:
No need for `main_args`. Just put `graph(get_args())` inside the main guard. See comment on `get_args` for updates on argument handling. Usually there's no reason to define a function for argument handling, just put that inside the main guard as well. The only reason to have it as a function would be if multiple scripts use it via `import`.

Reply:
I mainly defined a function so that pylint would stop insisting that my variable names should be in all caps.

Reply:
All caps is common and accepted practice. No need to define functions if they serve no purpose.

Reply:
Have removed the main_args function.